src/App.tsx

```diff
@@ -1,60 +1,131 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(users => users.id === todo.userId) || null,
+}));
+
+const usersFromServerFilter = usersFromServer
+  .find(users => todosFromServer
+    .find(todo => users.id === todo.userId)) || null;
 
 export const App = () => {
+  const [name, setName] = useState('');
+  const [userName, setUser] = useState(usersFromServerFilter);
+  const [todos, setTodos] = useState<Todo[]>([...preparedTodos]);
+
+  const [formKey, setFormKey] = useState(10);
+  const [isShowEmptyNameMessage, setErrorName] = useState(false);
+  const [isShowEmptyUserMessage, setErrorUser] = useState(false);
+  const [currentUserId, setUserId] = useState(0);
```

Comment:
```suggestion
  const [isErrorName, setIsErrorName] = useState(false);
  const [isErrorUser, setIsErrorUser] = useState(false);
  const [currentUserId, setCurrentUserId] = useState(0);
```
Try not to use so long names. Set function should be named according to the first name.
Also please pay attention to these rules:
![image](https://user-images.githubusercontent.com/96409563/201625630-e5629788-e070-493e-90a2-96c8a186b6c7.png)